### PR TITLE
Blog: A Post Mortem of The Multiple Counting Bug

### PR DESCRIPTION
**Pull request by dEBRUYNE-1**

### CODE REVIEW

File: _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+The second variant of the bug, [of which a detailed report can be found on](https://hackerone.com/reports/379049), entails the code not imposing a check against dummy transaction public keys. Therefore, a hacker could utilize the alternative transaction public keys feature to trick the wallet into scanning the outputs in a transaction twice. As a result, the receiving wallet would report that it had received two times the amount it had actually received. Similar to the first variant of the bug, the balance was not affected.
```
Comment (SamsungGalaxyPlayer) [inline _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md]: "of which a detailed report can be found on" -> "of which a detailed report can be found on HackerOne"

File: _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Now, unfortunately, this variant of the bug was fairly trivial to exploit, i.e., an attacker simply had to append `add_tx_pub_key_to_extra(tx, txkey_pub);` in `src/cryptonote_core/cryptonote_tx_utils.cpp` and his transaction public key would, after recompiling the code, be duplicated. Practically speaking this works as follows. The attackers starts with appending `src/cryptonote_core/cryptonote_tx_utils.cpp`, say, thrice^1, thereby creating four identical public transactions keys for his transactions. Subsequently, he sends a transaction of, say, 1 XMR to an exchange he seeks to exploit. The exchange will, most likely, credit the attacker with 4 XMR. The hacker, thereafter, withdraws this 4 XMR, thereby basically robbing the exchange of 3 XMR. If the exchange does not regularly check the balance against the sum of incoming and outgoing transfers or checks the hotwallet for any abnormalities, the hacker can basically repeat aforementioned process until the Monero hot wallet is emptied or, worst case scenario, all Monero funds are depleted.
```
Comment (SamsungGalaxyPlayer) [inline _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md]: "Now, unfortunately," -> "Unfortunately,"

Replace all instances of "his" with "their" and "he" with "they"

File: _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+Now, after v0.12.3.0 was tagged, I (and others) privately notified as many exchanges, services, and merchants as possible. Obviously, this is not the preferred method, as it (i) invariably excludes organizations that I (and others) personally do not have contact with, but are an essential part of the Monero ecosystem and (ii) may invoke a view of preferential treatment. In addition, the bug should have been reported on the public mailing list, but, alas, wasn't. An oversight which should be learnt from. Furthermore, we, as the Monero community, should seek improvements that would streamline this vulnerability report process (i.e. reporting a critical vulnerability to exchanges, services, and merchants). A "private" mailing list to which only services, merchants, and exchange are able to subscribe may be in line with this notion. Some kind of verification of subscribers would be required though, which could be a tedious process. Although, it would probably be more secure than a public mailing list to which a clever attacker would undoubtedly subscribe.
```
Comment (SamsungGalaxyPlayer) [inline _posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md]: "Now, after" -> "After"

Link the public mailing lists at lists.getmonero.org

**Comment by dEBRUYNE-1** [comment]
@SamsungGalaxyPlayer - Thanks, implemented suggestions.

**Pull request merged**